tests/testthat/test-derive_var_trtemfl.R

```diff
@@ -59,8 +95,26 @@ test_that("derive_var_trtemfl Test 2: with end_window and worsening", {
   )
 })
 
-## Test 3: considering trt end time ----
-test_that("derive_var_trtemfl Test 3: considering trt end time", {
+## Test 3: with end_window and worsening within grouping variable----
+test_that("derive_var_trtemfl Test 3: with end_window and worsening within grouping variable", {
```

Comment:
I feel like more tests should be written - that is a lot of new code under the hood that just has  one test backing it up!!  

What happens if group_var is specified and some of the other arguments are missing or poorly defined?

Reply:
Sure Ben, I will add some more tests.

Reply:
can we add the following data for testing - 
currently, there are some differences between `expected2`  and `comp`, do you agree with `TRTEMFL` values from `expected2`?
```r
expected2 <- tribble(
  ~USUBJID, ~ASTDTM, ~AENDTM, ~AEITOXGR, ~AETOXGR, ~AEGRPID, ~TRTEMFL,
  # starting before treatment and ending during treatment
  "1", "2021-12-30T20:00", "2022-01-14T11:00", "2", "3", "1", NA,
  "1", "2022-01-05T20:15", "2022-06-01T01:23", "2", "1", "1", NA,
  "1", "2022-01-10T20:15", "2022-01-11T01:23", "2", "2", "1", "Y",
  "1", "2022-01-13T20:15", "2022-03-01T01:23", "2", "1", "1", "Y",
  "1", "2021-12-30T20:00", "2022-01-14T11:00", "2", "2", "2", NA,
  "1", "2022-01-05T20:15", "2022-06-01T01:23", "2", "1", "2", NA,
  "1", "2022-01-10T20:15", "2022-03-01T01:23", "2", "1", "2", NA,
) %>%
  mutate(
    ASTDTM = lubridate::ymd_hm(ASTDTM),
    AENDTM = lubridate::ymd_hm(AENDTM),
    TRTSDTM = if_else(USUBJID == "1", lubridate::ymd_hm("2022-01-01T01:01"), ymd_hms("")),
    TRTEDTM = if_else(USUBJID == "1", lubridate::ymd_hm("2022-04-30T23:59"), ymd_hms(""))
  )

adae2 <- select(expected2, -starts_with("TRTEM"))

comp <- derive_var_trtemfl(
  adae2,
  new_var = TRTEMFL,
  trt_end_date = TRTEDTM,
  end_window = 10,
  initial_intensity = AEITOXGR,
  intensity = AETOXGR,
  group_var = AEGRPID
)
``` 

`comp`:
```r
> comp
# A tibble: 7 × 9
  USUBJID ASTDTM              AENDTM              AEITOXGR AETOXGR AEGRPID TRTSDTM             TRTEDTM             TRTEMFL
  <chr>   <dttm>              <dttm>              <chr>    <chr>   <chr>   <dttm>              <dttm>              <chr>  
1 1       2021-12-30 20:00:00 2022-01-14 11:00:00 2        3       1       2022-01-01 01:01:00 2022-04-30 23:59:00 NA     
2 1       2022-01-05 20:15:00 2022-06-01 01:23:00 2        1       1       2022-01-01 01:01:00 2022-04-30 23:59:00 Y      
3 1       2022-01-10 20:15:00 2022-01-11 01:23:00 2        2       1       2022-01-01 01:01:00 2022-04-30 23:59:00 Y      
4 1       2022-01-13 20:15:00 2022-03-01 01:23:00 2        1       1       2022-01-01 01:01:00 2022-04-30 23:59:00 Y      
5 1       2021-12-30 20:00:00 2022-01-14 11:00:00 2        2       2       2022-01-01 01:01:00 2022-04-30 23:59:00 NA     
6 1       2022-01-05 20:15:00 2022-06-01 01:23:00 2        1       2       2022-01-01 01:01:00 2022-04-30 23:59:00 Y      
7 1       2022-01-10 20:15:00 2022-03-01 01:23:00 2        1       2       2022-01-01 01:01:00 2022-04-30 23:59:00 Y  
```